### PR TITLE
fix(lsp): respect DENO_FUTURE for BYONM config

### PR DESCRIPTION
**Pull request by nayeemrmn**

### CODE REVIEW

File: tests/integration/lsp_tests.rs [hunk 1]
```diff
@@ -11786,6 +11786,45 @@ fn lsp_jupyter_byonm_diagnostics() {
...
+  client.initialize_default();
+  let diagnostics = client.did_open(json!({
+    "textDocument": {
+      "uri": temp_dir.uri().join("file.ts").unwrap(),
+      "languageId": "typescript",
+      "version": 1,
+      "text": "import \"npm:chalk\";\n",
+    },
+  }));
+  assert_eq!(
```
Comment (bartlomieju) [inline tests/integration/lsp_tests.rs:11804]: Can you update the test for check for a package that is actually listed in `package.json`?

File: tests/util/server/src/lsp.rs [hunk 2]
```diff
@@ -534,6 +549,9 @@ impl LspClientBuilder {
       .arg("lsp")
       .stdin(Stdio::piped())
       .stdout(Stdio::piped());
+    for (key, value) in &self.env_specified {
+      command.env(key, value);
+    }
```
Comment (dsherret) [inline tests/util/server/src/lsp.rs:554]: Oh, this probably should be inheriting the environment from the `TestContext` similar to how `new_command()` works. That said, this is good to have too.

File: tests/integration/lsp_tests.rs [hunk 3]
```diff
@@ -11786,6 +11786,56 @@ fn lsp_jupyter_byonm_diagnostics() {
...
+          },
+          "end": {
+            "line": 1,
+            "character": 26,
+          },
+        },
+        "severity": 1,
+        "code": "resolver-error",
+        "source": "deno",
+        "message": format!("Could not find a matching package for 'npm:chalk' in '{}/package.json'. You must specify this as a package.json dependency when the node_modules folder is not managed by Deno.", temp_dir.path()),
```
Comment (nayeemrmn) [inline tests/integration/lsp_tests.rs:11832]:
```suggestion
        "message": format!("Could not find a matching package for 'npm:chalk' in '{}'. You must specify this as a package.json dependency when the node_modules folder is not managed by Deno.", temp_dir.path().join("package.json")),
```

**Review by dsherret — APPROVED** [review]
LGTM!

**Pull request merged**